### PR TITLE
Update get_recommended_themes() to check filtered response for active themes

### LINKED ISSUE #40916: Customize Your Store: theme->is_active should be checked after filtering API response
Rather than rely on the API to tell the plugin what theme is active (via the `is_active` response value) we should let the server determine this for itself. Otherwise, we put the onus of knowing what theme is active on anyone who might be using this filter. It would be better to filter the response first, then populate the theme array with `is_active` values based on the response from internal theme functions. 

This issue will become more obvious when we filter the API response to get onboarding theme data from WooCommerce.com, which has no way of accurately returning correct `is_active` values without the requesting server sending it's active theme data along with the request. It makes more sense to validate this in the WC plugin instead of expecting the API to return correct values.

### PR DESCRIPTION
**Pull request by crunnells**
Updates get_recommended_themes() function so that it checks all of the filtered themes to see if any are currently active so we can display the badge on the front end.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40916.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Update WooCommerce with this branch, and make sure the `customize-store` feature flag is enabled.
2. Download [this gist](https://gist.github.com/crunnells/4940b3edf5377f796ffa0ed1d948fbd1) and install it as a plugin in wp-admin
3. Activate the uploaded `` plugin
4. Set the theme to Twenty Twenty Three
5. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store` and verify that Twenty Twenty Three shows the `Active theme` badge.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/API/OnboardingThemes.php [hunk 1]
```diff
@@ -260,7 +258,7 @@ public function get_recommended_themes( $request ) {
 				'color_palettes' => array(),
 				'total_palettes' => 0,
 				'slug'           => 'tsubaki',
-				'is_active'      => 'tsubaki' === $current_theme_slug,
+				'is_active'      => false,
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Admin/API/OnboardingThemes.php]: Nitpick: we could remove `is_active` from here now, to simplify things as it isn't the source of truth anymore.

**Review by adrianduffell — APPROVED** [review]
LGTM, thanks @crunnells 🚀 

I've left an optional suggestion. Pre-approving!

**Review by chihsuan — APPROVED** [review]
LGTM and tested well! 🚢

File: plugins/woocommerce/src/Admin/API/OnboardingThemes.php (hunk 1 above)
Comment (crunnells) [reply plugins/woocommerce/src/Admin/API/OnboardingThemes.php]: Good suggestion; I meant to remove them after testing to make sure we were overriding them correctly, and forgot to remove them later. Thanks!

**Pull request merged**